core/jvm/src/test/scala/com/ccadllc/cedi/dtrace/TypeclassLawTests.scala

```diff
@@ -177,6 +179,27 @@ class TypeclassLawTests extends FunSuite with Matchers with Checkers with Discip
     ()
   }
 
+  testAsync("parMap2 should be stack safe") { testC =>
+    implicit val cs = testC.contextShift[IO]
+    val count = 100000
+    val tasks = (0 until count).map(_ => TraceIO(1))
+    val sum = tasks.foldLeft(TraceIO(0))((acc, t) => (acc, t).parMapN(_ + _))
+    val f = sum.trace(tc).unsafeToFuture()
+    testC.tick()
+    f.value shouldBe Some(Success(count))
+    ()
+  }
+
+  testAsync("parTraverse should be stack safe") { testC =>
+    implicit val cs = testC.contextShift[IO]
+    val count = 100000
```

Comment:
maybe not a valid concern but one thing I'd mention for test purposes is maybe raising a number a bit? As one my machine with recommended stack options it gets up to 80k without a failure, not sure what jvm flags are set here for running tests though